OpenRA.Game/Server/Server.cs

```diff
@@ -783,6 +848,12 @@ public void ReceiveOrders(Connection conn, int frame, byte[] data)
 		{
 			if (frame == 0)
 				InterpretServerOrders(conn, data);
+
+			// HACK: This is not a good way to detect and relay sync hashes but it is currently the only way
+			else if (conn != null && data.Length != 0 && (OrderType)data[0] == OrderType.SyncHash)
+				DispatchOrdersToClients(conn, frame, data, true);
```

Comment:
Seeing that `DispatchOrdersToClients` ignores the buffer parameter:
```suggestion
				DispatchOrdersToClients(conn, frame, data);
```
I wonder if this is a bug?